R/utils_render_html.R

```diff
@@ -309,14 +292,23 @@ get_table_defs <- function(data) {
   if (table_width == "auto") {
 
     if (all(grepl("px", widths, fixed = TRUE))) {
+      # FIXME sometimes ends up being 0? #1532 and quarto-dev/quarto-cli#8233
       table_width <- "0px"
     } else if (all(grepl("%", widths, fixed = TRUE))) {
       table_width <- "100%"
     }
   }
 
   if (table_width != "auto") {
```

Comment:
@rich-iannone to fix #1532, should we just have this condition be
```suggestion
  if (!table_width %in% c("auto", "0", "0px")) {
```

If so, I will create a PR after this one is merged and update all snapshots.

I did the refactoring here to make the potential fix clearer in the future